lib/inspec/resources/oracledb_session.rb

```diff
@@ -69,9 +69,9 @@ def to_s
     private
 
     def pre_flight_check
-      return skip_resource "Option 'as_os_user' not available in Windows" if inspec.os.windows? && su_user
-      return fail_resource "Can't run Oracle checks without authentication" unless su_user && (user || password)
-      return fail_resource "You must provide a service name for the session" unless service
+      skip_resource "Option 'as_os_user' not available in Windows" if inspec.os.windows? && su_user
+      fail_resource "Can't run Oracle checks without authentication" unless su_user && (user || password)
+      fail_resource "You must provide a service name for the session" unless service
```

Comment:
I'm not entirely sure if it actually matters, but I think you want the `return`s so the first one wins rather than the last?

Reply:
I dont think I need returns at all on these methods. _all_ they do is set some instance variables.

https://github.com/inspec/inspec/blob/master/lib/inspec/plugin/v1/plugin_types/resource.rb#L136-L148

The runner ultimately checks these values and if the resource isn't failed or skipped it gets added.

https://github.com/inspec/inspec/blob/master/lib/inspec/runner.rb#L265-L274